jina/peapods/runtimes/jinad/client.py

```diff
@@ -103,17 +118,24 @@ def upload(self, dependencies: Sequence[str], workspace_id: str = None) -> str:
                     self.logger.error(f'fail to upload as {ex!r}')
 
     def create(self, args: 'Namespace') -> Optional[str]:
-        """ Create a remote pea/pod
-        :param args: the arguments in dict that pea can accept.
-                     (convert argparse.Namespace to Dict before passing to this method)
-        :return: the identity of the spawned pea/pod
+        """ Create a remote Pea / Pod
+
+        :param args: the arguments for remote Pea
+        :type args: Namespace
+        :raises requests.exceptions.RequestException:
+        :return: the identity of the spawned Pea / Pod
+        :rtype: Optional[str]
         """
+
         with ImportExtensions(required=True):
             import requests
 
         try:
             payload = replace_enum_to_str(vars(self._mask_args(args)))
-            r = requests.post(url=self.store_api, json=payload, timeout=args.timeout_ready if args.timeout_ready > 0 else self.timeout)
+            # set timeout to None if args.timeout_ready is -1 (wait forever)
+            r = requests.post(url=self.store_api,
+                              json=payload,
+                              timeout=args.timeout_ready if args.timeout_ready != -1 else None)
```

Comment:
@JoanFM changed https://github.com/jina-ai/jina/pull/1879 here.

Reply:
Can u make sure, that if `-1` is sent, and the Pea fails to start in the other end, the Exception is sent in response?

Reply:
If the remote Pea fails to start, we'll capture the exception in `r.json()`